src/Interpreter/Executors/Bash/Executor.php

```diff
@@ -37,36 +37,57 @@ public function capture(string $script, array $args = [], string $cwd = null, ar
             $script = substr_replace($script, 'echo -n ', $pos, 1);
         }
 
-        exec($this->buildCommand($script, $args, $cwd, $env), $output, $status);
+        $descriptorSpec = [
+            0 => ['pipe', 'r'],
+            1 => ['pipe', 'w'],
+            2 => STDERR,
+        ];
+
+        $pipes = [];
+        $process = proc_open($this->buildCommand($script, $args, $cwd), $descriptorSpec, $pipes, null, array_merge($_ENV, $env));
+
+        $output = '';
+        $status = 255;
+        if (is_resource($process)) {
+            fclose($pipes[0]);
+
+            $output = stream_get_contents($pipes[1]);
+            fclose($pipes[1]);
+
+            $status = proc_close($process);
+        }
 
         if ($status !== 0) {
             exit($status);
         }
 
-        return implode("\n", $output);
+        return $output;
     }
 
     public function getName(): string
     {
         return self::NAME;
     }
 
-    private function buildCommand(string $script, array $args, ?string $cwd, array $env): string
+    private function buildCommand(string $script, array $args, ?string $cwd): array
```

Comment:
not important, we should probably upgrade the PHPStan level but `/** @return list<string> */`